### PR TITLE
Allow remote servers to access server status page

### PR DESCRIPTION
**Pull request by mcarlton00**
### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [x] Changes to tests are appropriate (or tick if not required)
- [x] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [x] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Describe the changes you're proposing

Allow remote servers to read the Apache status page for monitoring purposes

### Pillar / config required to test the proposed changes

```
apache:
  server_status_access:
    ip:
      - 10.8.8.0/24
    host:
      - foo.example.com
```

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

File: apache/files/server-status.conf.jinja [hunk 1]
```diff
@@ -2,9 +2,19 @@
     SetHandler server-status
 {%- if apache.version == '2.4' %}
     Require local
+{%- for require_type,values in apache.get('server_status_require', {}).items() %}
```
Comment (myii) [inline apache/files/server-status.conf.jinja]: `server_status_require` or `server_status_access`?

File: apache/files/server-status.conf.jinja [hunk 2]
```diff
@@ -2,9 +2,19 @@
...
+{%- for value in values %}
+    Require {{ require_type }} {{ value }}
+{%- endfor %}
+{%- endfor %}
 {%- elif apache.version == '2.2' %}
     Order deny,allow
     Deny from all
     Allow from localhost
+{%- for require_type,values in apache.get('server_status_require', {}).items() %}
```
Comment (myii) [inline apache/files/server-status.conf.jinja]: `server_status_require` or `server_status_access`?

File: pillar.example [hunk 3]
```diff
@@ -385,3 +385,9 @@ apache:
     SSLProtocol: all -SSLv2 -SSLv3 -TLSv1
     SSLHonorCipherOrder: 'On'
     SSLOptions: "+StrictRequire"
+  server_status_access:
```
Comment (myii) [inline pillar.example]: `server_status_require` or `server_status_access`?

File: pillar.example [hunk 4]
```diff
@@ -385,3 +385,9 @@ apache:
...
+    ip:
+      - 10.8.8.0/24
+    host:
+      - foo.example.com
+
```
Comment (myii) [inline pillar.example]: This blank line needs to be removed to pass the `yamllint` check:

* https://travis-ci.com/saltstack-formulas/apache-formula/jobs/268961933#L363-L365

**Comment by myii** [comment]
@mcarlton00 This formula also uses `semantic-release` so the commit messages are used to evaluate the release.  Please see the following section for more info:

* https://github.com/saltstack-formulas/.github/blob/master/CONTRIBUTING.rst#commit-message-formatting

You can see how the current commit messages were evaluated by `commitlint`:

* https://travis-ci.com/saltstack-formulas/apache-formula/jobs/268961933#L430-L442

**Comment by mcarlton00** [comment]
I'd bounced back and forth between `_access` and `_require` and which one made more sense, and then wound up mixing them here.  Go figure.  Is there a preference of which one you think is clearer?

I'll merge the commits with the right message when I push the updates.

**Comment by myii** [comment]
> I'd bounced back and forth between `_access` and `_require` and which one made more sense, and then wound up mixing them here. Go figure. Is there a preference of which one you think is clearer?

@mcarlton00 Probably to go with the "`2.4`" option, i.e. `_require`.

---

I've simulated these changes in Travis and all looks to be working fine:

* https://travis-ci.org/myii/apache-formula/builds/627766020

So for [`2.4`](https://travis-ci.org/myii/apache-formula/jobs/627766022#L2263-L2270):

```conf
<Location "/server-status">
    SetHandler server-status
    Require local
    Require host foo.example.com
    Require host bar.example.com
    Require ip 10.8.8.0/24
    Require ip 10.10.0.0/16
</Location>
```

And for [`2.2`](https://travis-ci.org/myii/apache-formula/jobs/627766026#L3701-L3710):

```conf
<Location "/server-status">
    SetHandler server-status
    Order deny,allow
    Deny from all
    Allow from localhost
    Allow from foo.example.com
    Allow from bar.example.com
    Allow from 10.8.8.0/24
    Allow from 10.10.0.0/16
</Location>
```

**Comment by myii** [comment]
@mcarlton00 Just a minor change for that commit message:

```diff
feat(server-status): Allow remote servers to reach server-status page
feat(server-status): allow remote servers to reach server-status page
```

**Comment by myii** [comment]
Merged, thanks for the contribution @mcarlton00.  One thing we didn't cover here was any documentation changes.  Do you think it's worth adding a note under the following section?

* https://github.com/saltstack-formulas/apache-formula#apacheserver_status

**Comment by mcarlton00** [comment]
It probably should be adjusted some, yeah.  I didn't realize until now that it only says localhost there, which could be misleading at this point.  It still defaults to localhost only, but could be added to via the pillar, so I'm not sure what the best phrasing for it would be.

**Comment by myii** [comment]
@mcarlton00 Something based on your description above would be useful:

> Allow remote servers to read the Apache status page for monitoring purposes

Also a mention of the section in `pillar.example` to refer back to.

**Comment by mcarlton00** [comment]
Want a new PR for that?  afaik, updating a merged branch won't get us anywhere

**Comment by myii** [comment]
@mcarlton00 Sure, that's the right way to proceed.

**Pull request merged**